server/configuration-validation.js

```diff
@@ -1,125 +1,192 @@
-      collection: collection,
-    })
+      return this
+    },
+    validateMeilisearchSettings() {
+      // Settings is either undefined or an object
+      if (settings !== undefined && !isObject(settings)) {
+        log.error(`the "settings" option of "${name}" should be an object`)
+      } else if (settings !== undefined) {
+        options.settings = settings
+      }
+
+      return this
+    },
+    validatePopulateEntryRule() {
+      // PopulateEntry is either undefined or an object/array/string`
+      if (
+        populateEntryRule !== undefined &&
+        !isObject(populateEntryRule) &&
+        !Array.isArray(populateEntryRule) &&
+        typeof populateEntryRule !== 'string'
+      ) {
+        log.error(
+          `the "populateEntryRule" option of "${name}" should be an object/array/string`
+        )
+      } else if (populateEntryRule !== undefined) {
+        options.populateEntryRule = populateEntryRule
+      }
+
+      return this
+    },
+    validateNoInvalidKeys() {
+      // Keys that should not be present in the configuration
+      Object.keys(excedent).map(key => {
+        log.warn(`The attribute "${key}" of "${name}" is not a known option`)
+      })
+
+      return this
+    },
+    get() {
+      return options
+    },
   }
 }
 
-function validateCollectionConfiguration({ configuration, collection }) {
-  const validApiFields = [
-    'indexName',
-    'transformEntry',
-    'settings',
-    'filterEntry',
-    'populateEntryRule',
-  ]
+function PluginOptions(configuration) {
+  const log = strapi.log // has to be inside a scope
+  const { apiKey, host, ...collections } = configuration
+  const options = {}
 
-  if (configuration === undefined) {
-    return
-  }
+  return {
+    validateConfiguration() {
+      // Configuration must be an object
+      if (!isObject(configuration)) {
+        log.error(
+          'The `config` field in the Meilisearch plugin configuration must be of type object'
+        )
+      }
 
-  if (configuration !== undefined && !isObject(configuration)) {
-    strapi.log.error(`The collection "${collection}" should be of type object`)
-    return {}
-  }
+      return this
+    },
 
-  const { indexName } = configuration
-  if (
-    (indexName !== undefined && typeof indexName !== 'string') ||
-    indexName === ''
-  ) {
-    strapi.log.error(
-      `the "indexName" param of "${collection}" should be a non-empty string`
-    )
-    delete configuration.indexName
-  }
-  if (
-    configuration.transformEntry !== undefined &&
-    typeof configuration.transformEntry !== 'function'
-  ) {
-    strapi.log.error(
-      `the "transformEntry" param of "${collection}" should be a function`
-    )
-    delete configuration.transformEntry
-  }
+    validateApiKey() {
+      // apiKey is either undefined or a string
+      if (apiKey !== undefined && typeof apiKey !== 'string') {
+        log.error(
+          '`apiKey` should be a string in Meilisearch plugin configuration'
```

Comment:
You should check your error messages as sometimes it seems you use `"` and sometimes `.
Maybe it's wanted, but not sure.